### PR TITLE
[FEATURE] serve: Add '--simple-index' parameter

### PR DESCRIPTION
**Pull request by jkoenig134**
Showing hidden files and a detailed view is now possible via command line options.
Needs SAP/ui5-server#255 to work properly.

## Pull Request Checklist
- [x] Reviewed the [Contributing Guidelines](https://github.com/SAP/ui5-tooling/blob/master/CONTRIBUTING.md#-contributing-code)
    + Especially the [How to Contribute](https://github.com/SAP/ui5-tooling/blob/master/CONTRIBUTING.md#how-to-contribute) section 
- [x] [No merge commits](https://github.com/SAP/ui5-tooling/blob/master/docs/Guidelines.md#no-merge-commits)
- [x] [Correct commit message style](https://github.com/SAP/ui5-tooling/blob/master/docs/Guidelines.md#commit-message-style)

### CODE REVIEW

File: lib/cli/commands/serve.js [hunk 1]
```diff
@@ -25,6 +25,18 @@ serve.builder = function(cli) {
 			default: false,
 			type: "boolean"
 		})
+		.option("show-hidden", {
+			describe: "Show hidden files in the served index.",
```
Comment (RandomByte) [inline lib/cli/commands/serve.js]:
```suggestion
			describe: "Show hidden files in the server directory listing.",
```

File: lib/cli/commands/serve.js [hunk 2]
```diff
@@ -25,6 +25,18 @@ serve.builder = function(cli) {
...
+			default: false,
+			alias: "hidden",
+			type: "boolean"
+		})
+		.option("show-details", {
+			describe: "Show a detailed table view in the served index.",
```
Comment (RandomByte) [inline lib/cli/commands/serve.js]:
```suggestion
			describe: "Show a detailed table view in the server directory listing.",
```

File: lib/cli/commands/serve.js [hunk 3]
```diff
@@ -25,6 +25,18 @@ serve.builder = function(cli) {
 			default: false,
 			type: "boolean"
 		})
+		.option("show-hidden", {
+			describe: "Show hidden files in the served index.",
+			default: false,
```
Comment (RandomByte) [inline lib/cli/commands/serve.js]: As stated in https://github.com/SAP/ui5-server/pull/255#discussion_r341977950, we would rather have a default `true` here as that is the current behavior.

File: lib/cli/commands/serve.js [hunk 4]
```diff
@@ -25,6 +25,18 @@ serve.builder = function(cli) {
...
+			alias: "hidden",
+			type: "boolean"
+		})
+		.option("show-details", {
+			describe: "Show a detailed table view in the served index.",
+			default: false,
```
Comment (RandomByte) [inline lib/cli/commands/serve.js]: Default `true`, see above.

**Comment by jkoenig134** [comment]
@RandomByte i changed the defaults and descriptions

**Comment by jkoenig134** [comment]
When is the next release? And will my changes go online then?

**Comment by RandomByte** [comment]
Thanks! I'll merge the server change right away and this one probably tomorrow. I want to go over the naming of the CLI parameters with some colleagues. We'll probably also do a release then 👍

**Comment by jkoenig134** [comment]
@RandomByte thanks a lot. My idea about the naming is not perfect, i know. 
Maybe `hide-dotfiles` instead of `show-hidden` would fit better in the tooling?

**Comment by RandomByte** [comment]
Yeah, I also found `hide-dotfiles` being used by the lighttpd project: https://redmine.lighttpd.net/projects/lighttpd/wiki/Docs_ModDirlisting

**Comment by jkoenig134** [comment]
Feel free to change it, maintainer edit is active!

**Comment by RandomByte** [comment]
CC @matz3 @svbender

**Comment by RandomByte** [comment]
@jkoenig134 Sorry but due to absence of some colleagues it might take another day or two to get this merged and released

**Comment by jkoenig134** [comment]
No problem!

**Comment by matz3** [comment]
I don't really mind about the naming.
I rather wonder whether this is really required as a CLI option. For the UI5 use cases I would expect that dotfiles should always be shown, as we have several of them (.library, .theming, .theme, ...).

**Comment by RandomByte** [comment]
hey @jkoenig134, we discussed this again internally. As Matthias already wrote, UI5 comes with various dot files, so we wonder whether it actually makes sense to offer a switch to hide them.

Also, we wonder whether there is any demand in the non-detail directory listing.

Do you think you will use these parameters yourself or know of a use case where they would be helpful?

**Comment by jkoenig134** [comment]
Hey @RandomByte ! I totally see that point. The dotfile parameter is not necessary.
The non-detail parameter is. We are migrating from a grunt-Tooling where the non detail view was the default. And I think it also has a really good handling.

**Comment by RandomByte** [comment]
Fair enough. It pretty much boiled down to not having the dotfile parameter and renaming the one to activate the simplified "tile" view to `--simple-index`. I'll make the necessary changes now.

**Comment by jkoenig134** [comment]
Thanks. Sounds great.

**Comment by RandomByte** [comment]
Actually merged as https://github.com/SAP/ui5-cli/commit/ff2e84a738d0c8cb075e60b8477625a04f5bb7fe, GitHub UI struggled a bit with me composing the squash commit while the master got new commits 🤷🏼‍♂️

**Comment by jkoenig134** [comment]
Finally! Thanks guys!

**Pull request merged**